### DISCUSSION TITLE
Fritz/SkyPortal PR Process

### DISCUSSION THREAD

**Discussion by profjsb** [skyportal/skyportal#3097] [General]
This describes a proposed (and updated) process for the pull request process for Fritz and SkyPortal development. This serves as a clarification and expansion on the [Process Guidelines in SkyPortal’s Contributing section.](https://skyportal.io/docs/contributing.html#process-guidelines)

As a reminder, a PR must:

- provide a descriptive yet concise title
- describe in the comments the problem being addressed or the new functionality added. Ideally it should reference one or more issues being closed. The introduction of a substantially large change (touching front and back end, adding a new framework, etc.) should be discussed ahead of time
- Add screenshots and/or movies showing the changes, if relevant
- add or modify a test to cover any new or changed functionality.

### People

- front-end core dev: Stefan, Guy, Josh, Theo, Michael
- back-end core dev: Stefan, Guy, Josh, Theo, Michael

### Reviews

- All first-time committers should have reviews completed by at least two people
- Changes to other front-end pages may have only 1 review from a front-end “core dev”
- Changes to the database model (in`models/`) must have two reviews
- Changes to any other back-end may have only 1 review from a back-end “core dev”
- Big and complicated: more than 1 review
- If more expertise is (believed to be) needed, a single reviewer may request another review
- Docs/docstrings with small changes/corrections can be self-merged
- a PR should pass review by all who explicitly expressed interest in reviewing (if, of course, they do so within a reasonable period of time; that being 2 days after expressing interest).

### Merging

- Who
    
    A PR with a single reviewer should be merged after all their comments are addressed. A PR with more than one reviewer should be merged after the comments of all reviewers are addressed. The creator of the PR may self-merge after all positive reviews (Changes Approved) after 24 hour if one of the reviewers did not yet merge.
    
- When
    
    In general, a PR may only be merged after all checks return successfully (ie. all ✅). There are exceptions, however. A PR may be considered mergable if all of the tests related to the functionality change are ✅  but a few unrelated tests fail (some are more brittle than they ought to be): in this case the original author can make the claim in a comment that “test failures are unrelated” and the original reviewer agrees (ie. gives a thumbs up).

**Comment by stefanv** [comment]
I'm happy to add @Theodlz as another core frontend reviewer.

**Reply by Theodlz** [reply]
Thanks Stefan !

**Comment by stefanv** [comment]
A point that is not covered here: a PR should pass review by all who explicitly expressed interest in reviewing (if, of course, they do so within a reasonable period of time; that being, say, 2 (?) days after expressing interest).
